Sphinx docs generation

pd0wm opened the pull request:
https://docs.aks.comma.ai/

pd0wm committed:
add sphinx

pd0wm committed:
switch theme

ChadBailey reviewed (commented):
Very excited to pull this branch down and try it for myself! Consider my recommendations with a grain of salt or something to get you thinking of the possibilities. Perhaps you can come up with even better strategies for handling these things than I did.

ChadBailey committed:
Experiment: sphinx docs generation updated (#22708)

* moved build to root gitignore, added .gitkeep

* Improved makefile doc build process

- Removed auto-generated docs from source control
- Moved apidoc.sh into Makefile
- Removed make.bat (can add back if Windows support desired)
- Added sphinx viewcode and markdown extensions
- Added feature to source /docs in build, so any .rst file in /docs will
  override the respective file during the build process
- Added feature to copy all markdown/rst files from source into /build/
  during build process so they can be easily referenced while writing
  docs (see examples in index.md)
- Wrote basic starter index.md file

TODO: Add new dependencies to Pipfile [dev-packages]

* Revert accidental modification to Pipfile

pd0wm committed:
fix command substitution

pd0wm committed:
exclude xx

pd0wm committed:
improve docs

pd0wm committed:
dont include all docs in release build

pd0wm committed:
Add dockerfile

pd0wm committed:
update title

pd0wm committed:
include normal readme

pd0wm committed:
build container in CI

pd0wm committed:
use buildkit

pd0wm committed:
add login
